beater/beater.go

```diff
@@ -21,9 +23,18 @@ func New(b *beat.Beat, ucfg *common.Config) (beat.Beater, error) {
 	if err := ucfg.Unpack(beaterConfig); err != nil {
 		return nil, fmt.Errorf("Error reading config file: %v", err)
 	}
-
-	if b.Config != nil && b.Config.Output.Name() == "elasticsearch" {
-		beaterConfig.setElasticsearch(b.Config.Output.Config())
+	if beaterConfig.Frontend.isEnabled() {
+		_, err := regexp.Compile(beaterConfig.Frontend.LibraryPattern)
```

Comment:
I would prefer the one line assignment and nil check here, to limit `err`'s lifetime